### PR TITLE
Document luhn_check function

### LINKED ISSUE #7662: Document luhn_check function
Mention mod 10 and maybe link to https://en.wikipedia.org/wiki/Luhn_algorithm

Also not yet sure where that best goes. Update in list and examples needed.

### PR DESCRIPTION
**Pull request by findinpath**
Resolves #7662

### CODE REVIEW

File: docs/src/main/sphinx/functions/string.rst [hunk 1]
```diff
@@ -137,16 +150,16 @@ String functions
...
 
-        SELECT name 
+        SELECT name
         FROM nation
         WHERE SOUNDEX(name)  = SOUNDEX('CHYNA');
 
          name  |
         -------+----
-         CHINA | 
+         CHINA |
```
Comment (martint) [inline docs/src/main/sphinx/functions/string.rst:162]: Unrelated changes.

Comment (mosabua) [reply docs/src/main/sphinx/functions/string.rst:162]: Sure. They just remove whitespace so it wont matter .. do you still want @findinpath to revert those?

File: docs/src/main/sphinx/functions/string.rst [hunk 2]
```diff
@@ -100,6 +100,19 @@ String functions
     Tests whether a ``string`` of digits is valid according to the
     `Luhn algorithm <https://en.wikipedia.org/wiki/Luhn_algorithm>`_.
 
+    This checksum function also known as "modulo 10" or "mod 10" is
+    widely applied on most credit card numbers as well as government identification numbers
+    for distinguishing valid numbers from mistyped incorrect numbers.
+
+    Valid identification number::
+
+        select luhn_check('79927398713');-- true
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:109]: maybe move -- true to next line here and below

File: docs/src/main/sphinx/functions/string.rst [hunk 3]
```diff
@@ -100,6 +100,19 @@ String functions
     Tests whether a ``string`` of digits is valid according to the
     `Luhn algorithm <https://en.wikipedia.org/wiki/Luhn_algorithm>`_.
 
+    This checksum function also known as "modulo 10" or "mod 10" is
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:103]: use code highlighting maybe

```
``modulo 10``
```

Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:103]: also , before "also"

Comment (mosabua) [reply docs/src/main/sphinx/functions/string.rst:103]: and after mod 10

File: docs/src/main/sphinx/functions/string.rst [hunk 4]
```diff
@@ -100,6 +100,19 @@ String functions
...
+    widely applied on most credit card numbers as well as government identification numbers
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/string.rst:104]: widely applied on credit card numbers and government identification numbers to distinguish valid numbers from mistyped, incorrect numbers.

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/functions/string.rst (hunk 1 above)
Comment (findinpath) [reply docs/src/main/sphinx/functions/string.rst:162]: I'll just add these changes in a separate commit to keep the context of documenting `luhn_check` clear of unrelated changes. My IDE did these changes automatically and unfortunately I didn't notice them.

**Review by Damans227 — APPROVED** [review]
lgtm

File: docs/src/main/sphinx/functions/string.rst (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/functions/string.rst:104]: the suggestion was not applied yet .. any specific reason?

Comment (findinpath) [reply docs/src/main/sphinx/functions/string.rst:104]: I have misread your suggestion and applied only a part of it. Sorry for the misunderstanding.
I have changed now the sentence.

**Review by mosabua — APPROVED** [review]

**Review by Ordinant — APPROVED** [review]

**Comment by martint** [comment]
Thanks!
